tools/autograd/derivatives.yaml

```diff
@@ -1319,6 +1320,7 @@
 
 - name: transpose.int(Tensor(a) self, int dim0, int dim1) -> Tensor(a)
   self: grad.transpose(dim0, dim1)
+  result: auto_linear
```

Comment:
note that this will conflict with https://github.com/pytorch/pytorch/pull/59993
Not a big deal as the rebase will be simple, but we should not land them at the same time.